libqtile/window.py

```diff
@@ -159,6 +159,7 @@ def __init__(self, window, qtile):
         self.hidden = True
         self.group = None
         self.icons = {}
+        self.layout_state = {}
```

Comment:
Why not let each layout define what it needs, either a dictionary or something else?

```suggestion
        self.layout_state = None
```

Then you could just give the index directly for the xmonad layout, rather than a dictionary with just one key.

Reply:
Store object instead of a dictionary is an even better solution.